### ISSUE TITLE
different requirements in conda recipe and setup.py

### ISSUE THREAD

**Issue by xylar** [Unidata/cftime#59]
`cftime` seems to think it requires a different set of packages to run in the conda recipe:
```
    - python
    - setuptools
    - numpy x.x
```
and in `setup.py` (via `requirements.txt`):
```
numpy
cython
setuptools>=18.0
```
The requirement for `cython` appears to be retained in such a way that a conda environment that includes `cftime` but not `cython` and tries to use make use of `pkg_resources` gets the error:
```
...
pkg_resources.DistributionNotFound: The 'cython' distribution was not found and is required by cftime
```
The error is resolved by installing `cython` or including it as an (unnecessary) dependency of any conda package that depends on `cftime` but it would be more convenient if `cython` were only included in `setup.py` in `setup_requires` and not in `install_requires`.

So far, I haven't been able to create a super simple test that reproduces the issue.  I see it when I try to build a conda package from a repo that I'm just starting to develop.  Here is where I seem to need to include `cython`, though I don't think it should be needed:
https://github.com/xylar/misomip1analysis/blob/initial_stub/conda/recipe/meta.yaml#L28

**Comment by ocefpaf** [comment]
@xylar the conda-recipe here is only for Windows testing and I'm inclined to remove that altogether and use a bare bones AppVeoyr config. BTW, that is an issue with the recipe here and not `cftime`. (Due to its use of `python setup.py install` rather than `pip`.)

TL;DR you should use `cftime` from `conda-forge` if you need a conda package for that and not this recipe.

**Comment by xylar** [comment]
@ocefpaf, thank you very much for the reply.

I think I am, indeed, using `cftime` from `conda-forge`.  As with the recipe that you are removing here, `cython` is not included as a dependency (which I think is correct).  The problem seems to remain that `pkg_resources` is somehow picking up the list of dependencies passed to `setup.py` rather than the one that is included in the `conda-forge` recipe, meaning that it raises this erroneous `DistributionNotFound` exception.

Have you seen this kind of thing anywhere in the past?

@czender and colleagues saw exactly the same error at least once before.  I'm not sure how they managed to resolve it:
https://github.com/jaws/jaws/issues/115

**Comment by ocefpaf** [comment]
> I think I am, indeed, using cftime from conda-forge

OK. Let's move this discussion to the feedstock then. I'll check that in a moment...

**Comment by xylar** [comment]
@ocefpaf, I'm happy to move this over to `cftime-feedstock` if you're quite sure the problem is on the `conda-forge` side.  I still think the problem is most likely to be introduced in `setup.py` (regardless of whether `pip` is used or not).  If I do a grep on an environment where `cftime` has been installed from `conda-forge`, I see:
```
$ cat lib/python3.6/site-packages/cftime-1.0.0-py3.6.egg-info/requires.txt
numpy
cython
setuptools>=18.0
```
These requirements come directly from `requirements.txt` in this repo and are parsed in `setup.py`.  I have spent a bit over an hour trying to understand the intricacies of how `pkg_resources` parses requirements to no avail but I suspect it must be pulling information from the file above.

**Comment by ocefpaf** [comment]
I thought that, in `conda-forge` we where not cythinizing, and then we would not need it. But yes, you are correct. Sorry for the noise.

> have spent a bit over an hour trying to understand the intricacies of how pkg_resources 

`pkg_resources` is terrible, let me dig into this, we can probably be smarter on how we generate the cythonized files without introducing it as a dependency.

**Comment by xylar** [comment]
Thanks again for looking at this!

I really do think it could be as simple as removing `cython` from `install_requires` in `setup.py` but keeping it in `setup_requires`.  I don't think it's a problem that `cython` a build/setup requirement, it just shouldn't be an install requirement.  Maybe I'm missing something about how `setup.py` works...

**Comment by ocefpaf** [comment]
> I really do think it could be as simple as removing cython from install_requires in setup.py but keeping it in setup_requires. I don't think it's a problem that cython a build/setup requirement, it just shouldn't be an install requirement. Maybe I'm missing something about how setup.py works...

You are correct and that is exactly what #62 does :smile:

**Comment by xylar** [comment]
Excellent!  Thanks very much for taking care of this!
